PF/PF-Assignment2/ShoppingCart.java

```diff
@@ -0,0 +1,225 @@
+			this.itemName = itemName;
+			this.price = price;
+			this.quantity = quantity;
+			this.next = null;
+		}
+	}
+	
+	/*
+	 * @param   item id (int)
+	 * @return  item price
+	 */
+	public double getPrice(long itemID) {
+		Item item;
+		item = this.searchItem(itemID);
+		return item.price;
+	}
+	
+	/*
+	 * @param   item id (int), price of item (double)
+	 * @return  nothing
+	 * Update Item price after discount
+	 */
+	public void setPrice(long itemID, double price) {
+		Item item;
+		item = this.searchItem(itemID);
+		item.price = price;
+	}
+	
+	/*
+	 * @param   nothing
+	 * @return  true if Shopping Cart is Empty otherwise false
+	 */
+	public boolean isEmpty() {
+		if (this.firstItem == null)
+			return true;
+		return false;
+	}
+	
+	/*
+	 * @param   item name (string), item price (double), item quantity(long)
+	 * @return  nothing
+	 * Add item into the Shopping Cart list
+	 */
+	public void add(long itemID, String itemName, double price, long quantity) {
+		Item newItem = new Item(itemID, itemName, price, quantity);
+		total += price * (double)quantity;
+		if(isEmpty()) {
+			firstItem = newItem;
+			lastItem = newItem;
+		}
+		else {
+			lastItem.next = newItem;
+			lastItem = lastItem.next;
+		}
+	}
+	
+	/*
+	 * @param   Item id 
+	 * @return  nothing
+	 * Remove Item from list of Shopping cart
+	 */
+	public void remove(long itemID) {
+		Item it = firstItem;
+		Item pre = firstItem;
+		while(it != null) {
+			if(it.itemID == itemID)
+				break;
+			pre = it;
+			it = it.next;
+		}
+		if(it == firstItem) 
+			firstItem = firstItem.next;
+		else if(it == lastItem) {
+			lastItem = pre;
+			pre.next = null;
+		}
+		else 
+			pre.next = it.next;
+		System.out.println("Item id-" + it.itemID + " Removed !");
+		it = null;
+	}
+	
+	/*
+	 * @param   item-id (long)
+	 * @return  Item reference if it is found otherwise null
+	 */
+	public Item searchItem(long item_id) {
+		Item it = firstItem;
+		while(it != null) {
+			if(it.itemID == item_id)
+				return it;
+			it = it.next;
+		}
+		return null;
+	}
+	
+	/*
+	 * @param   item-id(long), change ("+" or "-")(String), Quantity(long)
+	 * @return  nothing
+	 * + used for Increase Quantity
+	 * - used for Decrease Quantity
+	 */
+	public void changeQuantity(long itemID, String change, long quantity) {
+		Item item = this.searchItem(itemID);
+		if (item == null)
+			System.out.println("Item is not found !");
+		else {
+			long newQuantity = (change.equals("+") ? quantity : quantity*(-1)) ;
+			long preQuantity = item.quantity;
+			item.quantity += newQuantity;
+			if(item.quantity < 0) {
+				newQuantity = preQuantity*(-1);
+				item.quantity = 0;
+			}
+			total += item.price * (double)newQuantity;
+			System.out.println("Item id-" + itemID + " Quantity is Changed !");
+			if(item.quantity == 0)
+				this.remove(item.itemID);
+		}
+	}
+	
+	
+	/*
+	 * @param   nothing
+	 * @return  nothing
+	 * Print list of all items present in Shopping Cart
+	 */
+	public void itemList() {
+		Item it = firstItem;
```

Comment:
Variable names should be meaningful. Build your habits from the very first day .